### DISCUSSION TITLE
Fluentbit to OTelcollector

### DISCUSSION THREAD

**Discussion by RitikaLaddha** [open-telemetry/opentelemetry-collector#10635] [Q&A]
Hi,
I want to upload logs from fluent bit to open telemetry collector on AKS cluster

So, I have installed otel collector as deployment [from here](https://artifacthub.io/packages/helm/opentelemetry-helm/opentelemetry-collector)
and fluentbit [from here](https://docs.fluentbit.io/manual/installation/kubernetes#installing-with-helm-chart)

This is my fluentbit configration for output to OTel

   ```
 [OUTPUT]
        Name                 opentelemetry
        Match                *
        Host                 http://my-opentelemetry-collector.namespace.svc.cluster.local
        Port                 4318
        Metrics_uri          /v1/metrics
        Logs_uri             /v1/logs
        Traces_uri           /v1/traces
        Log_response_payload True
        Tls                  On
        Tls.verify           Off
```

I have also added environment variable in fluentbit Daemonset
`env:
  - name: OTEL_EXPORTER_OTLP_TRACES_ENDPOINT
    value: http://my-opentelemetry-collector.namespace.svc.cluster.local:4318`

**But I am getting below error, in my fluentbit logs**

```
[error] [tls] error: unexpected EOF
[2024/07/17 13:51:23] [error] [output:opentelemetry:opentelemetry.1] no upstream connections available to my-opentelemetry-collector.namespace.svc.cluster.local:4317
[2024/07/17 13:51:23] [ warn] [engine] failed to flush chunk '1-1721224282.439304987.flb', retry in 7 seconds: task_id=8, input=tail.0 > output=opentelemetry.1 (out_id=1)
[2024/07/17 13:51:23] [ warn] [engine] failed to flush chunk '1-1721224282.536490801.flb', retry in 9 seconds: task_id=9, input=tail.0 > output=opentelemetry.1 (out_id=1)
```

Do I need to make any configuration change in Otel configmap?

**Comment by RitikaLaddha** [comment]
Hi [@open-telemetry/collector-maintainers](https://github.com/orgs/open-telemetry/teams/collector-maintainers)
Can anyone please guide me to achieve below flow? 
```
Logs -> Fluentbit -> OTel Collector -> Uptrace
                                    -> Jaeger
                                    -> etc..
```

**Reply by supersun** [reply]
https://opentelemetry.io/docs/specs/otel/logs/#third-party-application-logs

https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/fluentforwardreceiver
@RitikaLaddha I think this is what you are missing

**Comment by AndrewWinterman** [comment]
I saw this same error when i had tls on, but the otelcol was listening in plaintext. Try turning TLS off.
